src/benchmarks/micro/corefx/System.Text.Json/Perf.Utf8JsonReader.cs

```diff
@@ -0,0 +1,156 @@
+    public class Perf_Utf8JsonReader
+    {
+        [ParamsSource(nameof(DataGenerator))]
+        public JsonNumberSet NumberSet { get; set; }
+
+        public IEnumerable<JsonNumberSet> DataGenerator()
+        {
+            yield return new JsonNumberSet("positive numbers", "1234", "12345678901", "1234.5678", "1234.5678", "1234.5678");
+            
+            yield return new JsonNumberSet("positive numbers e2", "1234e2", "12345678901e2", "1234.5678e2", "1234.5678e2", "1234.5678e2");
+            yield return new JsonNumberSet("positive numbers e+2", "1234e+2", "12345678901e+2", "1234.5678e+2", "1234.5678e+2", "1234.5678e+2");
+            yield return new JsonNumberSet("positive numbers e-2", "1234e-2", "12345678901e-2", "1234.5678e-2", "1234.5678e-2", "1234.5678e-2");
+            
+            yield return new JsonNumberSet("positive numbers E2", "1234E2", "12345678901E2", "1234.5678E2", "1234.5678E2", "1234.5678E2");
+            yield return new JsonNumberSet("positive numbers E+2", "1234E+2", "12345678901E+2", "1234.5678E+2", "1234.5678E+2", "1234.5678E+2");
+            yield return new JsonNumberSet("positive numbers E-2", "1234E-2", "12345678901E-2", "1234.5678E-2", "1234.5678E-2", "1234.5678E-2");
+            
+            yield return new JsonNumberSet("negative numbers", "-1234", "-12345678901", "-1234.5678", "-1234.5678", "-1234.5678");
+            
+            yield return new JsonNumberSet("negative numbers e2", "-1234e2", "-12345678901e2", "-1234.5678e2", "-1234.5678e2", "-1234.5678e2");
+            yield return new JsonNumberSet("negative numbers e+2", "-1234e+2", "-12345678901e+2", "-1234.5678e+2", "-1234.5678e+2", "-1234.5678e+2");
+            yield return new JsonNumberSet("negative numbers e-2", "-1234e-2", "-12345678901e-2", "-1234.5678e-2", "-1234.5678e-2", "-1234.5678e-2");
+            
+            yield return new JsonNumberSet("negative numbers E2", "-1234E2", "-12345678901E2", "-1234.5678E2", "-1234.5678E2", "-1234.5678E2");
+            yield return new JsonNumberSet("negative numbers E+2", "-1234E+2", "-12345678901E+2", "-1234.5678E+2", "-1234.5678E+2", "-1234.5678E+2");
+            yield return new JsonNumberSet("negative numbers E-2", "-1234E-2", "-12345678901E-2", "-1234.5678E-2", "-1234.5678E-2", "-1234.5678E-2");
+        }
+
+        [Benchmark]
+        public int GetInt32Value() => GetInt32(NumberSet.int32Json);
+
+        [Benchmark]
+        public long GetInt64Value() => GetInt64(NumberSet.int64Json);
+
+        [Benchmark]
+        public float GetSingleValue() => GetSingleValue(NumberSet.floatJson);
+
+        [Benchmark]
+        public double GetDoubleValue() => GetDoubleValue(NumberSet.doubleJson);
+
+        [Benchmark]
+        public decimal GetDecimalValue() => GetDecimalValue(NumberSet.decimalJson);
+
+        private int GetInt32(ReadOnlySpan<byte> value)
+        {
+            var json = new Utf8JsonReader(value, isFinalBlock: true, state: default);
```

Comment:
If we are explicitly trying to measure how fast does the JsonReader reads and returns numbers specifically, I would try to minimize the overhead of creating a JsonReader and potentially even the loop checks. We can achieve this by having the reader read through a large array of (potentially random) numbers (so that the extra overhead is amortized).

Something like:
```C#
data = [number0, number1, number2, ..., numberN];
var json = new Utf8JsonReader(data, isFinalBlock: true, state: default);
while (json.Read());
```

OR

```C#
data = [number0, number1, number2, ..., numberN];
int result = 0;
var json = new Utf8JsonReader(data, isFinalBlock: true, state: default);
while (json.Read())
{
   // DO token type checks.
   json.TryGetInt32Value(out var currentInt);
   result |= currentInt;
}
return result;
```